pyproject.toml

```diff
@@ -0,0 +1,143 @@
+[build-system]
+requires = ['hatchling']
+build-backend = 'hatchling.build'
+
+[tool.hatch.version]
+path = 'pydantic/version.py'
+
+[project]
+name = 'pydantic'
+description = 'Data validation using Python type hints'
+authors = [
+    {name = 'Samuel Colvin', email = 's@muelcolvin.com'},
+    {name = 'Eric Jolibois', email = 'em.jolibois@gmail.com'},
+    {name = 'Hasan Ramezani', email = 'hasan.r67@gmail.com'},
+]
+license = {file = 'LICENSE'}
+readme = 'README.md'
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3 :: Only',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Intended Audience :: Developers',
+    'Intended Audience :: Information Technology',
+    'Intended Audience :: System Administrators',
+    'License :: OSI Approved :: MIT License',
+    'Operating System :: Unix',
+    'Operating System :: POSIX :: Linux',
+    'Environment :: Console',
+    'Environment :: MacOS X',
+    'Framework :: Hypothesis',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+    'Topic :: Internet',
+]
+requires-python = '>=3.7'
+dependencies = [ 'typing-extensions>=4.1.0' ]
+optional-dependencies = { email = ['email-validator>=1.0.3'], dotenv = ['python-dotenv>=0.10.4'] }
+dynamic = ['version']
+
+entry-points.hypothesis = {_ = 'pydantic._hypothesis_plugin'}
+
+[project.urls]
+Homepage = 'https://github.com/pydantic/pydantic'
+Documentation = 'https://pydantic-docs.helpmanual.io'
+Funding = 'https://github.com/sponsors/samuelcolvin'
+Source = 'https://github.com/pydantic/pydantic'
+Changelog = 'https://pydantic-docs.helpmanual.io/changelog'
+
+[tool.pytest.ini_options]
+testpaths = 'tests'
+filterwarnings = [
+    'error',
+    'ignore:path is deprecated.*:DeprecationWarning:certifi',
+]
+
+[tool.flake8]
+max-line-length = 120
+max-complexity = 14
+inline-quotes = 'single'
+multiline-quotes = 'double'
+ignore = ['E203', 'W503']
+per-file-ignores = [
+    'docs/examples/schema_unenforced_constraints.py:F811',
+    'docs/examples/validation_decorator_async.py:E402',
+    'docs/examples/types_constrained.py:F722',
+]
+
+[tool.coverage.run]
+source = ['pydantic']
+branch = true
+context = '${CONTEXT}'
+
+[tool.coverage.report]
+precision = 2
+exclude_lines = [
+    'pragma: no cover',
+    'raise NotImplementedError',
+    'raise NotImplemented',
+    'if TYPE_CHECKING:',
+    '@overload',
+]
+
+
+[tool.coverage.paths]
+source = [
+    'pydantic/',
+    '/Users/runner/work/pydantic/pydantic/pydantic/',
+    'D:\a\pydantic\pydantic\pydantic',
+]
+
+[tool.black]
+color = true
+line-length = 120
+target-version = ['py39']
```

Comment:
we could probably target 3.10 in the lint/format tools (3.11 is probably not yet supported)

Reply:
done, thanks.

Reply:
maybe we can do that for mypy (and maybe others I didn't check sorry)